app/models/signal_handler.rb

```diff
@@ -0,0 +1,31 @@
+class SignalHandler
+  def initialize
+    @read, @write = IO.pipe
+    Thread.new { run }
```

Comment:
JobQueue#clear locks a mutex, which isn't allowed inside signal handlers. This is also slightly better practice, since signal handlers are always preemptible (even if USR2 probably won't be called all that often).